integration/sync/test_collision-handlers.js

```diff
@@ -0,0 +1,115 @@
+      return done(err);
+    });
+  },
+  'afterEach': function(done) {
+    mongodb.dropCollection(DATASETID + '_collision');
+    dataHandlers = dataHandlersModule({
+      defaultHandlers: defaultDataHandlersModule(mongodb)
+    });
+    return done();
+  },
+  'test default create and list collisions': function(done) {
+    var collisionData = getCollisionData();
+    dataHandlers.handleCollision(DATASETID, {}, collisionData, function(err, res) {
+      assert.ok(!err);
+      dataHandlers.listCollisions(DATASETID, {}, function(err, res) {
+        var collision = _.values(res)[0];
+        assert.equal(1, _.size(res));
+        assert.equal(collisionData.hash, collision.hash);
+        assert.deepEqual(collisionData, collision);
+        done(err);
+      });
+    });
+  },
+  'test default removing collisions': function(done) {
+    var collisionData = getCollisionData();
+    dataHandlers.handleCollision(DATASETID, {}, collisionData, function(err, res) {
+      dataHandlers.listCollisions(DATASETID, {}, function(err, res) {
+        var collisionId = Object.keys(res)[0];
+        var collision = _.values(res)[0];
+        assert.equal(1, _.size(res));
+        dataHandlers.removeCollision(DATASETID, collisionId, {}, function(err, res) {
+          dataHandlers.listCollisions(DATASETID, {}, function(err, res) {
+            assert.ok(!err);
+            assert.equal(0, _.size(res));
+            return done();
+          });          
+        });
+      });
+    });
+  },
+  'test custom create collision': function(done) {
+    // Wrap the default handler, as long as we know a custom one is called.
+    var customHandler = sinon.spy(defaultHandlers, 'handleCollision');
+    dataHandlers.collisionHandler(DATASETID, customHandler);
+
+    var collisionData = getCollisionData();
+    dataHandlers.handleCollision(DATASETID, {}, collisionData, function(err) {
+      assert.ok(!err);
+      assert.ok(customHandler.calledOnce);
+      return done();
+    });
+  },
+  'test custom list collision': function(done) {
+    var customHandler = sinon.spy(defaultHandlers, 'listCollisions');
+    dataHandlers.listCollisionsHandler(DATASETID, customHandler);
+    dataHandlers.listCollisions(DATASETID, {}, function(err) {
+      assert.ok(!err);
+      assert.ok(customHandler.calledOnce);
+      return done();
+    });
+  },
+  'test custom remove collision': function(done) {
+    var customHandler = sinon.spy(defaultHandlers, 'removeCollision');
+    dataHandlers.removeCollisionHandler(DATASETID, customHandler);
+
+    var collisionData = getCollisionData();
+    dataHandlers.handleCollision(DATASETID, {}, collisionData, function(err, res) {
+      dataHandlers.removeCollision(DATASETID, res.uid, {}, function(err) {
+        assert.ok(!err);
+        dataHandlers.listCollisions(DATASETID, {}, function(err) {
+          assert.ok(!err);
+          assert.ok(customHandler.calledOnce);
+          return done();
+        });
+      });
+    });
+  }
```

Comment:
can we also add some tests to cover the `listCollisions` and `removeCollision` in 'sync/index.js' file as well? Mainly because that is the API we exposed and there should be some tests to cover those.